broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartition.java

```diff
@@ -430,16 +461,40 @@ private StreamProcessor createStreamProcessor(final ZeebeDb zeebeDb) {
     return closingPartitionFuture;
   }
 
-  private void tearDownBaseInstallation() {
+  private void tearDownBaseInstallation(final CompletableActorFuture<Void> closeFuture) {
     if (closeStateReplication()) {
+      closeFuture.complete(null);
       return;
     }
 
     if (closeSnapshotController()) {
```

Comment:
In these cases we also would not delete the log stream right?

Reply:
Yes. It is safe to call `closeLogStream` in all cases. I will update.